p2p/wire_codec.go

```diff
@@ -0,0 +1,234 @@
+package p2p
```

Comment:
Having a custom wire codec makes me very uncomfortable.

Let's take this offline on Friday's protocol hour